File: tasks/3536-maximum-product-of-two-digits.ts
// Accepted 1088 / 1088 testcases passed Sergey Pomortsev submitted at Mar 23, 2026 22:27
// Solution Runtime 1 ms Beats 100.00% Memory 58.05 MB Beats 100.00%

function maxProduct(n: number): number {
  return n.toString().split('').sort((a, b) => b > a ? 1 : -1).slice(0, 2).reduce((acc, curr) => acc * +curr, 1);
};

File: tasks/3536-maximum-product-of-two-digits.go
// Accepted 1088 / 1088 testcases passed Sergey Pomortsev submitted at Jul 02, 2026 15:21
// Runtime 0 ms Beats 100.00% Memory 4.02 MB Beats 60.87%

import "strconv"

func maxProduct(n int) int {
  var str = strconv.Itoa(n)
  var max int
  var max2 int

  for _, r := range str {
    var num = int(r - '0')

    if num >= max {
      max2 = max
      max = num
    } else if num < max && num > max2 {
      max2 = num
    }
  }

  return max * max2
}